src/command/trade.rs

```diff
@@ -352,37 +343,13 @@ fn handle_network_event(
     ethereum_wallet: Arc<ethereum::Wallet>,
     bitcoin_connector: Arc<comit::btsieve::bitcoin::BitcoindConnector>,
     ethereum_connector: Arc<comit::btsieve::ethereum::Web3Connector>,
+    take_request_handler: F,
```

Comment:
Why do you pass in the logic for processing the take request as a function from the outside?

Reply:
Because depending on whether I am in `trade` or `wrap-up` then I would process the take request differently.